e2e/ansible/playbooks/utils/deploy_task.yml

```diff
@@ -0,0 +1,9 @@
+---
+- name: Deploy application files
+  shell: source ~/.profile; kubectl apply -f {{ item }}
+  args:
+    executable: /bin/bash
+  delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
+  with_items: "{{ app_yml }}"
```

Comment:
Let us add a failed_when case here which looks for "resource/object" created - this will take care of spec errors (I suspect this may already be the case if deploying an error-prone spec might result in non-zero retcode..)